io.c

```diff
@@ -3377,69 +3397,77 @@ io_write_nonblock(rb_execution_context_t *ec, VALUE io, VALUE str, VALUE ex)
 
 /*
  *  call-seq:
- *     ios.read([length [, outbuf]])    -> string, outbuf, or nil
+ *    read(maxlen = nil)             -> new_string or nil
+ *    read(maxlen = nil, out_string) -> out_string or nil
  *
- *  Reads _length_ bytes from the I/O stream.
+ *  Reads (in binary mode) up to +maxlen+ bytes from the stream.
  *
- *  _length_ must be a non-negative integer or +nil+.
+ *  If +maxlen+ is +nil+, reads the entire stream and returns the content as a string;
+ *  encoding conversion is applied, if applicable:
  *
- *  If _length_ is a positive integer, +read+ tries to read
- *  _length_ bytes without any conversion (binary mode).
- *  It returns +nil+ if an EOF is encountered before anything can be read.
- *  Fewer than _length_ bytes are returned if an EOF is encountered during
- *  the read.
- *  In the case of an integer _length_, the resulting string is always
- *  in ASCII-8BIT encoding.
+ *    f = File.new('t.txt')
+ *    f.read # => "This is line one.\nThis is the second line.\nThis is the third line.\n"
+ *    f.eof  # => true
+ *    f.read # => ""
  *
- *  If _length_ is omitted or is +nil+, it reads until EOF
- *  and the encoding conversion is applied, if applicable.
- *  A string is returned even if EOF is encountered before any data is read.
+ *  If +maxlen+ is zero, returns an empty string:
  *
- *  If _length_ is zero, it returns an empty string (<code>""</code>).
+ *    f = File.new('t.txt')
```

Comment:
I think we can drop this example, it's a corner case and can be mentioned at the end.

Reply:
Text moved, example dropped.